README.md
<h1 align="center" id="heading"> Select a Proxmox Helper </h1>

<p align="center"><sub> Always remember to use due diligence when sourcing scripts and automation tasks from third-party sites. </sub></p>

<a href="https://github.com/primitiv0/Proxmox/blob/main/LICENSE"><img src="https://badgen.net/github/license/primitiv0/Proxmox" ></a> <a href="https://primitiv0.github.io/Proxmox/"><img src="https://badgen.net/github/checks/primitiv0/Proxmox/main/"></a> <a href="https://github.com/primitiv0/Proxmox/discussions"><img src="https://github.com/primitiv0/Proxmox/blob/main/misc/images/discussions.png?raw=true" height="21" /></a> <a href="https://github.com/primitiv0/Proxmox/blob/main/CHANGELOG.MD"><img src="https://github.com/primitiv0/Proxmox/blob/main/misc/images/change.png?raw=true" height="21" /></a>🔸

<details>
<summary markdown="span"> Proxmox VE 7 Post Install</summary>
 
<p align="center"><img src="https://github.com/home-assistant/brands/blob/master/core_integrations/proxmoxve/icon.png?raw=true" height="100"/></p>

<h1 align="center" id="heading"> Proxmox VE 7 Post Install </h1>

The script will give options to Disable the Enterprise Repo, Add/Correct PVE7 Sources, Enable the No-Subscription Repo, Add Test Repo, Disable Subscription Nag and Update Proxmox VE.
 
Run the following in the Proxmox Shell. ⚠️ **PVE7 ONLY**

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/post-install-v3.sh)"
```

It's recommended to answer `y` to all options.

____________________________________________________________________________________________ 

</details>

<details>
<summary markdown="span"> Proxmox Kernel Clean</summary>
 
<p align="center"><img src="https://github.com/home-assistant/brands/blob/master/core_integrations/proxmoxve/icon.png?raw=true" height="100"/></p>

<h1 align="center" id="heading">Proxmox Kernel Clean </h1>

Cleaning unused kernel images is not only good because of a reduced grub menu, but also gains some disk space.
 
Run the following in the Proxmox Shell.

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/kernel-clean-v3.sh)"
```
____________________________________________________________________________________________ 

</details>

<details>
<summary markdown="span"> Proxmox Edge Kernel Tool</summary>
 
<p align="center"><img src="https://github.com/home-assistant/brands/blob/master/core_integrations/proxmoxve/icon.png?raw=true" height="100"/></p>

<h1 align="center" id="heading">Proxmox Edge Kernel Tool </h1>

Proxmox [Edge Kernels](https://github.com/fabianishere/pve-edge-kernel) are custom Linux Kernels for Proxmox VE 7. Keeping up with new Kernel releases instead of LTS

Run the following in the Proxmox Shell.

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/edge-kernel.sh)"
```
____________________________________________________________________________________________ 

</details>

<details>
<summary markdown="span"> Proxmox CPU Scaling Governor</summary>
 
<p align="center"><img src="https://github.com/primitiv0/Proxmox/blob/main/misc/images/cpu.png?raw=true" height="100"/></p>

<h1 align="center" id="heading">Proxmox CPU Scaling Governor </h1>

CPU Scaling Governor enables the operating system to scale the CPU frequency up or down in order to save power or improve performance.

[Generic Scaling Governors](https://www.kernel.org/doc/html/latest/admin-guide/pm/cpufreq.html?#generic-scaling-governors)
 
Run the following in the Proxmox Shell.

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/scaling-governor.sh)"
```
____________________________________________________________________________________________ 

</details>

<details>
<summary markdown="span"> Proxmox LXC Updater</summary>
 
<p align="center"><img src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Felpuig.xeill.net%2FMembers%2Fvcarceler%2Farticulos%2Fcontenedores-con-lxd-lxc%2Fcontainers.png&f=1&nofb=1" height="100"/></p>

<h1 align="center" id="heading">Proxmox LXC Updater </h1>

Update All LXC's Fast & Easy
 
Run the following in the Proxmox Shell.

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/update-lxcs.sh)"
```
____________________________________________________________________________________________ 

</details>

<details>
<summary markdown="span"> Proxmox Dark Theme</summary>
 
<p align="center"><img src="https://camo.githubusercontent.com/f6f33a09f8c1207dfb3dc1cbd754c2f3393562c11b1c999751ad9a91a656834a/68747470733a2f2f692e696d6775722e636f6d2f536e6c437948462e706e67" height="100"/></p>

<h1 align="center" id="heading"> Proxmox Discord Dark Theme </h1>

A dark theme for the Proxmox Web UI by [Weilbyte](https://github.com/Weilbyte/PVEDiscordDark)
 
Run the following in the Proxmox Shell.

```yaml
bash <(curl -s https://raw.githubusercontent.com/Weilbyte/PVEDiscordDark/master/PVEDiscordDark.sh ) install
```

To uninstall the theme, simply run the script with the `uninstall` command.

____________________________________________________________________________________________ 

</details>


<details>
<summary markdown="span"> 🔸Home Assistant OS VM</summary>
 
<p align="center"><img src="https://github.com/primitiv0/Proxmox/blob/main/misc/images/haos.png?raw=true"/></p>
 
<h1 align="center" id="heading"> Home Assistant OS VM </h1>
<h3 align="center"> Option to create VM using the Latest or Stable Image </h3>

The script automates the manual process of finding, downloading and extracting the Official KVM (qcow2) disk image provided by the Home Assistant Team, creating a VM with user defined settings, importing and attaching the disk, setting the boot order and starting the VM. No hidden (kpartx, unzip, ect...) installs of any kind. Supports lvmthin, zfspool, nfs, dir and btrfs storage types.

To create a new Proxmox Home Assistant OS VM, run the following in the Proxmox Shell

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/vm/haos-vm-v3.sh)"
```
<h3 align="center" id="heading">⚡ Default Settings:  4GB RAM - 32GB Storage - 2vCPU - Stable Image⚡</h3>
 
After the script completes, click on the VM, then on the **_Summary_** tab to find the VM IP.

**Home Assistant Interface - IP:8123**

____________________________________________________________________________________________ 
 
</details>


<details>
<summary markdown="span"> Home Assistant Container LXC </summary>
 
<p align="center"><img src="https://www.docker.com/sites/default/files/d8/2019-07/vertical-logo-monochromatic.png" alt="Docker Logos | Docker" width="100" height="100"/>
<img src="https://avatars.githubusercontent.com/u/13844975?s=200&amp;v=4" alt="@home-assistant" width="100" height="100"/><img src="https://avatars1.githubusercontent.com/u/22225832?s=400&amp;v=4" alt="GitHub - portainer/portainer-docs: Portainer documentation" width="100" height="100"/></p>

<h1 align="center" id="heading"> Home Assistant Container LXC </h1>

A standalone container-based installation of Home Assistant Core

To create a new Proxmox Home Assistant Container LXC, run the following in the Proxmox Shell.
 
```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/ct/homeassistant-v3.sh)"
```
 
<h3 align="center" id="heading">⚡ Default Settings:  2GB RAM - 16GB Storage - 2vCPU ⚡</h3>

**Home Assistant Interface - IP:8123**

**Portainer Interface - IP:9000**

⚙️ **Path to HA /config**
```yaml
/var/lib/docker/volumes/hass_config/_data
 ```
⚙️ **To Edit the HA configuration.yaml** (Recommend Using File Browser)
 
Run in the LXC console
```yaml
nano /var/lib/docker/volumes/hass_config/_data/configuration.yaml
```
Save and exit the editor with “Ctrl+O”, “Enter” and “Ctrl+X”

⚙️ **Copy Data From a Existing Home Assistant LXC to another Home Assistant LXC**

Run in the Proxmox Shell
```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/ha-copy-data.sh)"
 ```

⚙️ **To Allow USB Device Passthrough:**
 
Run in the Proxmox Shell. (**replace `106` with your LXC ID**)
```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/usb-passthrough.sh)" -s 106
```
 
Reboot the LXC to apply the changes


⚙️ **To Install HACS:**

Run in the LXC console
```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/hacs.sh)"
```
After install, reboot Home Assistant and **clear browser cache** then Add HACS integration.


⚙️ [**Update Menu**](https://raw.githubusercontent.com/primitiv0/Proxmox/main/misc/images/update-menu.png)

Run in the LXC console
```yaml
./update
```
____________________________________________________________________________________________ 
</details>

<details>
<summary markdown="span"> Podman Home Assistant Container LXC </summary>
 
<p align="center"><img src="https://heise.cloudimg.io/width/223/q50.png-lossy-50.webp-lossy-50.foil1/_www-heise-de_/imgs/18/2/5/8/2/8/1/0/podman_logo-670078d7ea1d15a6.png" width="100" height="100"/>
<img src="https://avatars.githubusercontent.com/u/13844975?s=200&amp;v=4" alt="@home-assistant" width="100" height="100"/><img/><img src="https://raw.githubusercontent.com/SelfhostedPro/Yacht/master/readme_media/Yacht_logo_1_dark.png" height="80"/><img/></p>
 
<h1 align="center" id="heading"> Podman Home Assistant Container LXC </h1>
A standalone container-based installation of Home Assistant Core

⚠️ Podman seems to need a privileged LXC

To create a new Proxmox Podman Home Assistant Container LXC, run the following in the Proxmox Shell. 

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/ct/podman-homeassistant-v3.sh)"
```
<h3 align="center" id="heading">⚡ Default Settings:  2GB RAM - 16GB Storage - 2vCPU ⚡</h3>
 
**Home Assistant Interface - IP:8123**
 
**Yacht Interface - IP:8000**

⚙️ **Path to HA /config**
```yaml
/var/lib/containers/storage/volumes/hass_config/_data
 ```
⚙️ **To edit the HA configuration.yaml**
 
Run in the LXC console
```yaml
nano /var/lib/containers/storage/volumes/hass_config/_data/configuration.yaml
```
Save and exit the editor with “Ctrl+O”, “Enter” and “Ctrl+X”

⚙️ **Copy Data From a Existing Home Assistant LXC to a Podman Home Assistant LXC**

Run in the Proxmox Shell
```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/ha-copy-data-podman.sh)"
 ```

⚙️ **To allow USB device passthrough:**
 
Run in the Proxmox Shell. (**replace `106` with your LXC ID**)
```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/usb-passthrough.sh)" -s 106
```
 
Reboot the LXC to apply the changes

⚙️ **To Install HACS:**

Run in the LXC console
```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/podman_hacs.sh)"
```
After install, reboot Home Assistant and **clear browser cache** then Add HACS integration.

⚙️ **Initial Yacht Login**

**username** 
 ```yaml
 admin@yacht.local
 ```
 **password** 
 ```yaml
 pass
 ```

____________________________________________________________________________________________ 
</details>

<details>
<summary markdown="span"> ioBroker LXC</summary>
 
<p align="center"><img src="https://github.com/ioBroker/ioBroker/blob/master/img/logos/ioBroker_Logo_256px.png?raw=true" height="100"/></p>

<h1 align="center" id="heading"> ioBroker LXC </h1>
 
[ioBroker](https://www.iobroker.net/#en/intro) is an open source automation platform.
 
To create a new Proxmox ioBroker LXC, run the following in the Proxmox Shell.

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/ct/iobroker-v3.sh)"
```

<h3 align="center" id="heading">⚡ Default Settings:  2GB RAM - 8GB Storage - 2vCPU ⚡</h3>
 
**ioBroker Interface - IP:8081**

⚙️ **To Update ioBroker**

```yaml
update from the ioBroker UI
```

____________________________________________________________________________________________ 
 
</details>

<details>
<summary markdown="span"> openHAB LXC</summary>
 
<p align="center"><img src="https://www.openhab.org/openhab-logo-square.png?raw=true" height="100"/></p>

<h1 align="center" id="heading"> openHAB LXC </h1>
 
[openHAB](https://www.openhab.org/), a vendor and technology agnostic open source automation software for your home.
 
To create a new Proxmox openHAB LXC, run the following in the Proxmox Shell.

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/ct/openhab-v3.sh)"
```

<h3 align="center" id="heading">⚡ Default Settings:  2GB RAM - 8GB Storage - 2vCPU ⚡</h3>
 
**openHAB Interface - IP:8080**

⚙️ **To Update openHAB**

```yaml
apt update && apt upgrade -y
```

____________________________________________________________________________________________ 
 
</details>

<details>
<summary markdown="span"> Homebridge LXC</summary>
 
<p align="center"><img src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fwww.dingz.ch%2Fadmin%2Fdata%2Ffiles%2Fintegration%2Flogo%2F20%2F200514-em-logo-homebridge_logo.png%3Flm%3D1589459081&f=1&nofb=1" height="100"/></p>

<h1 align="center" id="heading"> Homebridge LXC </h1>

[Homebridge](https://homebridge.io/) allows you to integrate with smart home devices that do not natively support HomeKit

To create a new Proxmox Homebridge LXC, run the following in the Proxmox Shell.

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/ct/homebridge-v3.sh)"
```

<h3 align="center" id="heading">⚡ Default Settings:  1GB RAM - 4GB Storage - 1vCPU ⚡</h3>
 
**Homebridge Interface - IP:8581**

⚙️ **Initial Login**

**username** 
 ```yaml
 admin
 ```
 **password** 
 ```yaml
 admin
 ```
Config File Path	`/var/lib/homebridge/config.json`
 
Storage Path	`/var/lib/homebridge`
 
Restart Command	`sudo hb-service restart`
 
Stop Command	`sudo hb-service stop`
 
Start Command	`sudo hb-service start`
 
View Logs Command	`sudo hb-service logs`
 
Systemd Service File	`/etc/systemd/system/homebridge.service`
 
Systemd Env File	`/etc/default/homebridge`

⚙️ **To Update Homebridge**

```yaml
Update from the Homebridge UI
```

 ___________________________________________________________________________________________ 
 
</details>



<details>
<summary markdown="span"> ESPHome LXC</summary>
 
<p align="center"><img src="https://github.com/home-assistant/brands/blob/master/core_integrations/esphome/dark_icon@2x.png?raw=true" height="100"/></p>

<h1 align="center" id="heading"> ESPHome LXC </h1>

[ESPHome](https://esphome.io/) is a system to control your ESP8266/ESP32 by simple yet powerful configuration files and control them remotely through Home Automation systems.

To create a new Proxmox ESPHome LXC, run the following in the Proxmox Shell.

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/ct/esphome-v3.sh)"
```

<h3 align="center" id="heading">⚡ Default Settings:  1GB RAM - 4GB Storage - 2vCPU ⚡</h3>
 
**ESPHome Interface - IP:6052**

⚙️ **To Update ESPHome**

Run in the LXC console
```yaml
pip3 install esphome --upgrade
```

____________________________________________________________________________________________ 
 
</details>



<details>
<summary markdown="span"> Nginx Proxy Manager LXC </summary>
 
<p align="center"><img src="https://nginxproxymanager.com/logo.png" alt="hero" height="100"/></p>


<h1 align="center" id="heading"> Nginx Proxy Manager LXC </h1>

[Nginx Proxy Manager](https://nginxproxymanager.com/) Expose your services easily and securely

To create a new Proxmox Nginx Proxy Manager LXC Container, run the following in the Proxmox Shell.

```yaml
 bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/ct/nginx-proxy-manager-v3.sh)"
```
<h3 align="center" id="heading">⚡ Default Settings:  1GB RAM - 3GB Storage - 1vCPU ⚡</h3>

____________________________________________________________________________________
 
Forward port `80` and `443` from your router to your Nginx Proxy Manager LXC IP.

Add the following to your `configuration.yaml` in Home Assistant.
```yaml
 http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 192.168.100.27 ###(Nginx Proxy Manager LXC IP)###
```

**Nginx Proxy Manager Interface - IP:81**
 
⚙️ **Initial Login**

**username** 
 ```yaml
 admin@example.com
 ```
 **password** 
 ```yaml
 changeme
 ```
⚙️ **To Update Nginx Proxy Manager**

Run in the LXC console
```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/npm_update.sh)"
```

 ____________________________________________________________________________________________ 

</details>

<details>
<summary markdown="span"> MQTT LXC</summary>
 
<p align="center"><img src="https://mosquitto.org/images/mosquitto-text-side-28.png" height="75"/></p>

<h1 align="center" id="heading"> MQTT LXC </h1>

[Eclipse Mosquitto](https://mosquitto.org/) is an open source message broker that implements the MQTT protocol

To create a new Proxmox MQTT LXC, run the following in the Proxmox Shell.

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/ct/mqtt-v3.sh)"
```

<h3 align="center" id="heading">⚡ Default Settings:  512MiB RAM - 2GB Storage - 1vCPU ⚡</h3>
 
Mosquitto comes with a password file generating utility called mosquitto_passwd.
```yaml
sudo mosquitto_passwd -c /etc/mosquitto/passwd <usr>
```
Password: < password >

Create a configuration file for Mosquitto pointing to the password file we have just created.
```yaml
sudo nano /etc/mosquitto/conf.d/default.conf
```
This will open an empty file. Paste the following into it.
```yaml
allow_anonymous false
persistence true
password_file /etc/mosquitto/passwd
listener 1883
```
Save and exit the text editor with "Ctrl+O", "Enter" and "Ctrl+X".

Now restart Mosquitto server.
```yaml
sudo systemctl restart mosquitto
```

⚙️ **To Update MQTT:**

Run in the LXC console
```yaml
apt update && apt upgrade -y
```

____________________________________________________________________________________________ 
 
</details>

<details>
<summary markdown="span"> Node-Red LXC </summary>
 
<p align="center"><img src="https://github.com/home-assistant/brands/blob/master/custom_integrations/nodered/icon.png?raw=true" height="100"/></p>

<h1 align="center" id="heading"> Node-Red LXC </h1>
 
[Node-RED](https://nodered.org/) is a programming tool for wiring together hardware devices, APIs and online services in new and interesting ways.

To create a new Proxmox Node-RED LXC, run the following in the Proxmox Shell.

```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/ct/node-red-v3.sh)"
```

<h3 align="center" id="heading">⚡ Default Settings:  1GB RAM - 4GB Storage - 1vCPU ⚡</h3>
 
**Node-Red Interface - IP:1880**
 
⚙️ **To Restart Node-Red:**

Run in the LXC console
```yaml
node-red-restart
```

⚙️ **To Update Node-Red:**

Run in the LXC console (Restart after update)
```yaml
npm install -g --unsafe-perm node-red
```

⚙️ **To Install Node-Red Themes** ⚠️ **Backup your flows before running this script!!**

Run in the LXC console
```yaml
bash -c "$(wget -qLO - https://github.com/primitiv0/Proxmox/raw/main/misc/node-red-themes.sh)"
```

____________________________________________________________________________________________ 
 
</details>

<details>
<summary markdown="span"> 🔸n8n LXC</summary>
 
<p align="center"><img src="https://docs.n8n.io/_images/n8n-docs-icon.svg" height="70"/></p>


CHANGELOG.MD
<a href="https://primitiv0.github.io/Proxmox/"><img src="https://github-readme-stats.vercel.app/api?username=primitiv0&theme=blue-green" height="130"/></a>

# Change Log
All notable changes to this project will be documented in this file.

## 2022-07-24

### Changed

- **Home Assistant OS VM**
  - Present the drive to the guest as a solid-state drive rather than a rotational hard disk. There is no requirement that the underlying storage actually be backed by SSD's. 
  - When the VM’s filesystem marks blocks as unused after deleting files, the SCSI controller will relay this information to the storage, which will then shrink the disk image accordingly.


## 2022-07-22

### Changed

- **n8n LXC** (thanks to @cyakimov)
  - NEW Script

## 2022-07-21

### Changed

- **grocy LXC**
  - NEW Script

## 2022-07-17

### Changed

- **Vaultwarden LXC**
  - NEW Vaultwarden Update (post 2022-05-29 installs only) Script
  - NEW Web-vault Update (any) Script

## 2022-07-14

### Changed

- **MagicMirror Server LXC**
  - NEW Script

## 2022-07-13

### Changed

- **Proxmox Edge Kernel Tool**
  - NEW Script

## 2022-07-11

### Changed

- **Home Assistant OS VM**
  - Supports lvmthin, zfspool, nfs, dir and btrfs storage types.

## 2022-07-08

### Changed

- **openHAB LXC**
  - NEW Script

## 2022-07-03

### Changed

- **Tailscale**
  - NEW Script

## 2022-07-01

### Changed

- **Home Assistant OS VM**
  - Allow different storage types (lvmthin, nfs, dir).

## 2022-06-30

### Changed

- **Prometheus LXC**
  - NEW Script

## 2022-06-06

### Changed

- **Whoogle LXC**
  - NEW Script

## 2022-05-29

### Changed

- **Vaultwarden LXC**
  - Code refactoring
- **CrowdSec**
  - NEW Script

## 2022-05-21

### Changed

- **Home Assistant OS VM**
  - Code refactoring

## 2022-05-19

### Changed

- **Keycloak LXC**
  - NEW Script

## 2022-05-18

### Changed

- **File Browser**
  - NEW Script

## 2022-05-13

### Changed

- **PostgreSQL LXC**
  - NEW Script

## 2022-05-10

### Changed

- **deCONZ LXC**
  - NEW Script

## 2022-05-07

### Changed

- **NocoDB LXC**
  - ADD update script

## 2022-05-06

### Changed

- **PhotoPrism LXC**
  - ADD GO Dependencies for full functionality

## 2022-05-05

### Changed

- **Ubuntu LXC**
  - ADD option to define version (18.04 20.04 21.10 22.04)

## 2022-04-28

### Changed

- **v3 Script**
  - Remove Internet Check

## 2022-04-27

### Changed

- **Home Assistant OS VM**
  - ADD Option to set Bridge, VLAN and MAC Address
- **v3 Script**
  - Improve Internet Check (prevent ‼ ERROR 4@57)

## 2022-04-26

### Changed

- **Home Assistant OS VM**
  - Fixed bad path
  - ADD Option to create VM using Latest or Stable image
- **UniFi Network Application LXC**
  - ADD Local Controller Option

## 2022-04-25

### Changed

- **v3 Script**
  - Improve Error Handling

## 2022-04-23

### Changed

- **v3 Script**
  - ADD Internet Connection Check
- **Proxmox VE 7 Post Install**
  - NEW v3 Script
- **Proxmox Kernel Clean**
  - NEW v3 Script

## 2022-04-22

### Changed

- **Omada Controller LXC**
  - Update script to install version 5.1.7
- **Uptime Kuma LXC**
  - ADD Update script

## 2022-04-20

### Changed

- **Ubuntu LXC**
  - ADD option to install version 20.04 or 21.10
- **v3 Script**
  - ADD option to set Bridge

## 2022-04-19

### Changed

- **ALL LXC's**
  - New [V3 Install Script](https://github.com/primitiv0/Proxmox/issues/162) 
- **ioBroker LXC**
  - New Script V3

## 2022-04-13

### Changed

- **Uptime Kuma LXC**
  - New Script V2

## 2022-04-11

### Changed

- **Proxmox LXC Updater**
  - ADD option to skip stopped containers
- **Proxmox VE 7 Post Install**
  - ADD PVE 7 check

## 2022-04-10

### Changed

- **Debian 11 LXC**
  - ADD An early look at the v3 install script

## 2022-04-09

### Changed

- **NocoDB LXC**
  - New Script V2

## 2022-04-05

### Changed

- **MeshCentral LXC**
  - New Script V2

## 2022-04-01

### Changed

- **Scripts** (V2)
  - FIX Pressing enter without making a selection first would cause an Error 

## 2022-03-28

### Changed

- **Docker LXC**
  - Add Docker Compose Option (@wovalle)

## 2022-03-27

### Changed

- **Heimdall Dashboard LXC**
  - New Update Script

## 2022-03-26

### Changed

- **UniFi Network Application LXC**
  - New Script V2
- **Omada Controller LXC**
  - New Script V2

## 2022-03-25

### Changed

- **Proxmox CPU Scaling Governor**
  - New Script


## 2022-03-24

### Changed

- **Plex Media Server LXC**
  - Switch to Ubuntu 20.04 to support HDR tone mapping
- **Docker LXC**
  - Add Portainer Option

## 2022-03-23

### Changed

- **Heimdall Dashboard LXC**
  - New Script V2

## 2022-03-20

### Changed

- **Scripts** (V2)
  - ADD choose between Automatic or Manual DHCP  

## 2022-03-18

### Changed

- **Technitium DNS LXC**
  - New Script V2
- **WireGuard LXC**
  - Add WGDashboard

## 2022-03-17

### Changed

- **Docker LXC**
  - New Script V2

## 2022-03-16

### Changed

- **PhotoPrism LXC**
  - New Update/Branch Script

## 2022-03-15

### Changed

- **Dashy LXC**
  - New Update Script

## 2022-03-14

### Changed

- **Zwavejs2MQTT LXC**
  - New Update Script

## 2022-03-12

### Changed

- **PhotoPrism LXC**
  - New Script V2

## 2022-03-11

### Changed

- **Vaultwarden LXC**
  - New V2 Install Script

## 2022-03-08

### Changed

- **Scripts** (V2)
  - Choose between Privileged or Unprivileged CT and Automatic or Password Login 
- **ESPHome LXC**
  - New V2 Install Script
- **Zwavejs2MQTT LXC**
  - New V2 Install Script
- **Motioneye LXC**
  - New V2 Install Script
- **Pihole LXC**
  - New V2 Install Script
- **GamUntu LXC**
  - New V2 Install Script

## 2022-03-06

### Changed

- **Zwavejs2MQTT LXC**
  - New GUI script to copy data from one Zwavejs2MQTT LXC to another Zwavejs2MQTT LXC

## 2022-03-05

### Changed

- **Homebridge LXC**
  - New Script V2

## 2022-03-04

### Changed

- **Proxmox Kernel Clean**
  - New Script

## 2022-03-03

### Changed

- **WireGuard LXC**
  - New Script V2

## 2022-03-02

### Changed

- **Proxmox LXC Updater**

ct/plex-v1.sh
#!/usr/bin/env bash

while true; do
    read -p "This will create a New Plex Media Server LXC Container. Proceed(y/n)?" yn
    case $yn in
        [Yy]* ) break;;
        [Nn]* ) exit;;
        * ) echo "Please answer yes or no.";;
    esac
done
set -o errexit  
set -o errtrace
set -o nounset  
set -o pipefail 
shopt -s expand_aliases
alias die='EXIT=$? LINE=$LINENO error_exit'
trap die ERR
CHECKMARK='\033[0;32m\xE2\x9C\x94\033[0m'
trap cleanup EXIT

function error_exit() {
  trap - ERR
  local DEFAULT='Unknown failure occured.'
  local REASON="\e[97m${1:-$DEFAULT}\e[39m"
  local FLAG="\e[91m[ERROR] \e[93m$EXIT@$LINE"
  msg "$FLAG $REASON"
  [ ! -z ${CTID-} ] && cleanup_ctid
  exit $EXIT
}
function warn() {
  local REASON="\e[97m$1\e[39m"
  local FLAG="\e[93m[WARNING]\e[39m"
  msg "$FLAG $REASON"
}
function info() {
  local REASON="$1"
  local FLAG="\e[36m[INFO]\e[39m"
  msg "$FLAG $REASON"
}
function msg() {
  local TEXT="$1"
  echo -e "$TEXT"
}
function cleanup_ctid() {
  if [ ! -z ${MOUNT+x} ]; then
    pct unmount $CTID
  fi
  if $(pct status $CTID &>/dev/null); then
    if [ "$(pct status $CTID | awk '{print $2}')" == "running" ]; then
      pct stop $CTID
    fi
    pct destroy $CTID
  elif [ "$(pvesm list $STORAGE --vmid $CTID)" != "" ]; then
    pvesm free $ROOTFS
  fi
}
function cleanup() {
  popd >/dev/null
  rm -rf $TEMP_DIR
}
function load_module() {
  if ! $(lsmod | grep -Fq $1); then
    modprobe $1 &>/dev/null || \
      die "Failed to load '$1' module."
  fi
  MODULES_PATH=/etc/modules
  if ! $(grep -Fxq "$1" $MODULES_PATH); then
    echo "$1" >> $MODULES_PATH || \
      die "Failed to add '$1' module to load at boot."
  fi
}
TEMP_DIR=$(mktemp -d)
pushd $TEMP_DIR >/dev/null

wget -qL https://raw.githubusercontent.com/primitiv0/Proxmox/main/setup/plex_setup.sh

load_module overlay

while read -r line; do
  TAG=$(echo $line | awk '{print $1}')
  TYPE=$(echo $line | awk '{printf "%-10s", $2}')
  FREE=$(echo $line | numfmt --field 4-6 --from-unit=K --to=iec --format %.2f | awk '{printf( "%9sB", $6)}')
  ITEM="  Type: $TYPE Free: $FREE "
  OFFSET=2
  if [[ $((${#ITEM} + $OFFSET)) -gt ${MSG_MAX_LENGTH:-} ]]; then
    MSG_MAX_LENGTH=$((${#ITEM} + $OFFSET))
  fi
  STORAGE_MENU+=( "$TAG" "$ITEM" "OFF" )
done < <(pvesm status -content rootdir | awk 'NR>1')
if [ $((${#STORAGE_MENU[@]}/3)) -eq 0 ]; then
  warn "'Container' needs to be selected for at least one storage location."
  die "Unable to detect valid storage location."
elif [ $((${#STORAGE_MENU[@]}/3)) -eq 1 ]; then
  STORAGE=${STORAGE_MENU[0]}
else
  while [ -z "${STORAGE:+x}" ]; do
    STORAGE=$(whiptail --title "Storage Pools" --radiolist \
    "Which storage pool you would like to use for the container?\n\n" \
    16 $(($MSG_MAX_LENGTH + 23)) 6 \
    "${STORAGE_MENU[@]}" 3>&1 1>&2 2>&3) || exit
  done
fi
info "Using '$STORAGE' for Storage Location."

CTID=$(pvesh get /cluster/nextid)
info "Container ID is $CTID."

echo -e "${CHECKMARK} \e[1;92m Updating LXC Template List... \e[0m"
pveam update >/dev/null
echo -e "${CHECKMARK} \e[1;92m Downloading LXC Template... \e[0m"
OSTYPE=ubuntu
OSVERSION=${OSTYPE}-20.04
mapfile -t TEMPLATES < <(pveam available -section system | sed -n "s/.*\($OSVERSION.*\)/\1/p" | sort -t - -k 2 -V)
TEMPLATE="${TEMPLATES[-1]}"
pveam download local $TEMPLATE >/dev/null ||
  die "A problem occured while downloading the LXC template."

STORAGE_TYPE=$(pvesm status -storage $STORAGE | awk 'NR>1 {print $2}')
case $STORAGE_TYPE in
  dir|nfs)
    DISK_EXT=".raw"
    DISK_REF="$CTID/"
    ;;
  zfspool)
    DISK_PREFIX="subvol"
    DISK_FORMAT="subvol"
    ;;
esac
DISK=${DISK_PREFIX:-vm}-${CTID}-disk-0${DISK_EXT-}
ROOTFS=${STORAGE}:${DISK_REF-}${DISK}

echo -e "${CHECKMARK} \e[1;92m Creating LXC Container... \e[0m"
DISK_SIZE=8G
pvesm alloc $STORAGE $CTID $DISK $DISK_SIZE --format ${DISK_FORMAT:-raw} >/dev/null
if [ "$STORAGE_TYPE" == "zfspool" ]; then
  warn "Some containers may not work properly due to ZFS not supporting 'fallocate'."
else
  mkfs.ext4 $(pvesm path $ROOTFS) &>/dev/null
fi
ARCH=$(dpkg --print-architecture)
HOSTNAME=plex
TEMPLATE_STRING="local:vztmpl/${TEMPLATE}"
pct create $CTID $TEMPLATE_STRING -arch $ARCH -features nesting=1 \
  -hostname $HOSTNAME -net0 name=eth0,bridge=vmbr0,ip=dhcp -onboot 1 -cores 2 -memory 2048\
  -ostype $OSTYPE -rootfs $ROOTFS,size=$DISK_SIZE -storage $STORAGE >/dev/null

LXC_CONFIG=/etc/pve/lxc/${CTID}.conf
cat <<EOF >> $LXC_CONFIG
lxc.cgroup2.devices.allow: c 226:0 rwm
lxc.cgroup2.devices.allow: c 226:128 rwm
lxc.cgroup2.devices.allow: c 29:0 rwm
lxc.mount.entry: /dev/fb0 dev/fb0 none bind,optional,create=file
lxc.mount.entry: /dev/dri dev/dri none bind,optional,create=dir
lxc.mount.entry: /dev/dri/renderD128 dev/renderD128 none bind,optional,create=file
EOF

MOUNT=$(pct mount $CTID | cut -d"'" -f 2)
ln -fs $(readlink /etc/localtime) ${MOUNT}/etc/localtime
pct unmount $CTID && unset MOUNT

echo -e "${CHECKMARK} \e[1;92m Starting LXC Container... \e[0m"
pct start $CTID
pct push $CTID plex_setup.sh /plex_setup.sh -perms 755
pct exec $CTID /plex_setup.sh

IP=$(pct exec $CTID ip a s dev eth0 | sed -n '/inet / s/\// /p' | awk '{print $2}')
info "Successfully Created Plex Media Server LXC to $CTID."
echo -e "\e[1;92m Plex Media Server should be reachable by going to the following URL.
             http://${IP}:32400/web
\e[0m"
